README.md

```diff
@@ -162,27 +162,37 @@ const resolvers = {
 
 # Typescript
 
-As the library cannot know the structure of your user object you need to add this to the global scope by writing your own "somename.d.ts" you can adapt the following template with whatever user variables that you want have added to the PassportRequest:
+As the library cannot know the structure of your user object you need to add this to the global scope by writing your own "somename.d.ts" you can add how your context definition should look like:
 
 ```ts
-declare module "graphql-passport" {
-  interface PassportRequest {
-    user?: {
-      username: string;
-    };
+import { Request as ExpressRequest } from 'express';
+import { PassportSubscriptionContext, PassportContext } from 'graphql-passport';
+import { MyUserObject } = 'some_user_file';
+
+export interface MyContext extends PassportContext<MyUserObject, ExpressRequest>{
+  dataSources: {
```

Comment:
This means that the MyContext type will also include a field `dataSources.myOtherAPI...` right? Where is the relation to `MyUserObject`?

Reply:
The `dataSources` has nothing to do with the user object. *Data sources* is one of those hidden gems (see https://www.apollographql.com/docs/apollo-server/data/data-sources/#accessing-data-sources-from-resolvers) that I wish that that I had found a while ago and that probably should always be a part of a custom context definition. Through it you can get access to your database or whatever you need in your resolver without all the annoying '../../models/' - now all you need to import is your context definition and if you can be certain that all your resolvers use the right model object.

Reply:
Ok cool, I didn't know that. Thanks for the link. But not everyone will be using data sources. So this will confuse people who don't know about that. I'll add a proposal for documentation in another comment and resolve the conversation here